src/coreclr/jit/importer.cpp

```diff
@@ -520,10 +520,9 @@ void Compiler::impAppendStmt(Statement* stmt, unsigned chkLevel, bool checkConsu
 
                 assert(retBuf->TypeIs(TYP_I_IMPL, TYP_BYREF));
 
-                GenTreeLclVarCommon* lclNode = retBuf->IsLocalAddrExpr();
-                if (lclNode != nullptr)
+                if (retBuf->OperIs(GT_LCL_VAR_ADDR))
```

Comment:
Can this not be `GT_LCL_FLD_ADDR`?

Reply:
No, only `LCL_VAR_ADDR`, currently at least.

It reflects the tight coupling this code has with what the importer actually produces for `stloc`. We don't look for indirect stores here for the same reason.